tests/PHPStan/Rules/Classes/data/bug-5333.php

```diff
@@ -0,0 +1,34 @@
+<?php declare(strict_types = 1);
+
+namespace Bug5333;
+
+use function PHPStan\Testing\assertNativeType;
+
+class Route {}
+
+class HelloWorld
+{
+	/**
+	 * @param Route|Route[] $foo
+	 *
+	 * @return Route
+	 **/
+	public function sayHello($foo): Route
+	{
+		assertNativeType('mixed', $foo);
+		if (\is_array($foo)) {
+			assertNativeType('array', $foo);
+			$res = $foo[0];
+			if (!$res instanceof Route) {
```

Comment:
The most interesting thing to me is the native type of `$res` before this condition.